python/truelayer_signing/utils.py

```diff
@@ -21,6 +25,35 @@ class HttpMethod(str, Enum):
 M = TypeVar("M", HttpMethod, Optional[HttpMethod])
 
 
+@dataclass(frozen=True)
+class JwsHeader:
+    alg: str
+    kid: str
+    tl_version: str
+    tl_headers: str
+    jku: Optional[str] = None
+
+    @classmethod
+    def from_dict(cls, header: Mapping[str, str]) -> "JwsHeader":
+        if any(
+            x not in header.keys() for x in ["alg", "kid", "tl_version", "tl_headers"]
+        ):
+            raise TlSigningException("Invaild Header")
+
+        if header["alg"] != "ES512":
```

Comment:
As the keys `alg` and `tl_version` are always expected to be `"ES512"` and `"2"`, perhaps consider an `Enum`

Reply:
I think creating a new type for two string literals is a bit much as they don't have any variations; I have refactored to use const instead `SIGNING_ALGORITHM and TL_VERSION`

Reply:
I think I meant that because they shouldn't have variations, we can use an enum here so it's explicit as well e.g.


```
class JwsHeaderFields(Enum):
    tl_version: str = "2"
    alg: str = "ES512"

...

if header["alg"] != JwsHeaderFields.alg:

...

```

What do you think?

Reply:
This way, you can also use this `Enum` in the unit tests

Reply:
I don't think that's an appropriate use of an `Enum`, but I do see what you're trying to do; how about this?
```python
class DefaultJwsHeaderFieldsValues:
  tl_version: str = TL_VERSION # if a new version comes out we can introduce a enum for the tl_version 
  alg: str = SIGNING_ALGORITHM # if we start to support new algorithms we can introduce a enume 
```

Reply:
That makes sense!